web-components/chart/chart.tsx

```diff
@@ -212,11 +193,56 @@ export class CobChart {
   componentDidLoad() {
     // Once the component loads, we initialize the view.
     this.view.initialize(`#${this.chartID}`);
-    // Attach tooltips to the chart
+
+    // Attach tooltips to the chart.
     vegaTooltip(this.view);
-    // If we are using selections, we build the dropdown.
+
+    // We grab the div the chart is sitting it and hold onto it as
+    // we'll use it in multiple places.
+    this.chartDiv = document.getElementById(this.chartID)!;
+
     if (this.config.selection) {
+      // If we are using selections, we build the dropdown.
       this.buildDropDownSelection();
+
+      // We set the view the run, but wait to populate specific pieces
+      // until after the data has loaded.
+      this.view.runAsync().then(() => {
+        // We grab the dataset we're charting. We do this here because
+        // it allows us to use Vega transforms then use the transformed data
+        // to populate our selections. It also means we don't have to
+        // fetch our data twice - Vega does it once and we leverage it.
+        this.dataset = this.view.data(this.config.data.name);
+        // We use the dataset to get our list of dropdown options.
+        this.selectOptions = Array.from(
```

Comment:
Be on the lookout for the need to polyfill this on older IE. Not sure if Stencil will have done it for us.